[gohjieling834] iP

gohjieling834 opened the pull request.

gohjieling834 committed:
Add Greet and Exit lines

- Added greeting and exit lines
- Renamed Chatbot name to Berry

gohjieling834 committed:
Add Greet/Exit lines and rename chatbot

- Added greeting and exit lines
- Renamed chatbot from Duke to Berry
- Renamed file Duke.java to Berry.java

gohjieling834 committed:
Add method 'echo'

The method 'echo' echos commands entered by the user, and exists when the user types the command 'bye'.

gohjieling834 committed:
Update echo method

Moved the variables and while loop in the echo method into main.

gohjieling834 committed:
Refactor echo method to addTask and add list method

Renamed and updated the echo method so that the method is able to
add tasks to the array, and the new method name reflects its functionality better.

gohjieling834 committed:
Add Task class

Added a Task class to represent tasks and encapsulate their related data.

gohjieling834 committed:
Update unmarkAsDone method

Included the line to update the isDone to be false.

gohjieling834 committed:
Add toggleTaskStatus method

Added the toggleTaskStatus method so that the user can mark/unmark the task.

Updated the addTask and list method to work with Task objects instead of Strings.

gohjieling834 committed:
Update code to comply with coding standard

saheer17 commented on src/main/java/Berry.java:
```diff
@@ -0,0 +1,57 @@
+    public static void main(String[] args) {
+        String chatbotName = "Berry";
+        String userCommand;
+        Task[] tasks = new Task[100];   //array of Task objects
```
Great that you used plural form for tasks, which is a collection of objects! Instead of using the number "100", which is a magic number, you can add a int variable named "taskCapacity" and initialise it as 100, then use that variable when initialising your array.

saheer17 commented on src/main/java/Berry.java:
```diff
@@ -0,0 +1,57 @@
+
+public class Berry {
+    public static void main(String[] args) {
+        String chatbotName = "Berry";
```
Since "Berry" is a constant, you can mark the variable as "final" and for constants, change the name of the variable such that it is in UPPER_SNAKE_CASE form.

saheer17 commented on src/main/java/Berry.java:
```diff
@@ -0,0 +1,57 @@
+                numberOfTasks++;
+            }
+            userCommand = in.nextLine();
+        }
```
For better readability, consider replacing the chained if–else statements with a switch on userCommand. This makes the command handling easier to read and easier to extend if new commands are added later

saheer17 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,37 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
The variables are declared as "protected", which can expose them and break encapsulation. You may want to declare them as private instead

saheer17 reviewed (commented):
Overall, Java coding standards were mostly followed, especially in the case of naming variables. You may want to consider using switch cases more to improve readability!

yujie-o commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,37 @@
+        isDone = true;
+    }
+
+    public void unmarkAsDone() {
```
```suggestion
    public void markAsUndone() {
```
Prefer intention-revealing methods for state changes. For naming, use verbs for actions.

yujie-o commented on src/main/java/Berry.java:
```diff
@@ -0,0 +1,57 @@
+            System.out.println(i + 1 + ".[" + tasks[i].getStatusIcon() + "] " + tasks[i].getDescription());
+        }
+        System.out.println("‧⋆ ✧˚₊‧⋆. ✧˚₊‧⋆‧‧⋆ ✧˚₊‧⋆. ✧˚₊‧⋆‧\n");
+    }
```
Good use of spaces around operators & in string concatenations.

yujie-o commented on src/main/java/Berry.java:
```diff
@@ -0,0 +1,57 @@
+
+    public static void printList(Task[] tasks, int numberOftasks) {
+        System.out.println("\n‧⋆ ✧˚₊‧⋆. ✧˚₊‧⋆‧‧⋆ ✧˚₊‧⋆. ✧˚₊‧⋆‧");
+        for (int i = 0; i < numberOftasks; i++) {
```
```suggestion
        for (int i = 0; i < numberOfTasks; i++) {
```
The variable numberOftasks doesn’t follow Java naming conventions. Variable names should be in camelCase (so it should be numberOfTasks), and in this case, taskCount or simply size would be clearer and more consistent with Java’s collection classes.

yujie-o commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,37 @@
+        isDone = done;
+    }
+
+    public String getStatusIcon() {
```
The isDone boolean and getStatusIcon() are simple but effective abstractions for tracking task state.

yujie-o reviewed (commented):
Overall, this is a solid start with good structure, a few adjustments around naming, validation, and documentation will make the code much more robust and aligned with Java coding standards.

gohjieling834 committed:
Add Todo, Deadline, Event classes

Added Todo, Deadline, Event classes as subclasses of Task.
This helps to support different task types.

gohjieling834 committed:
Extract helper methods and add task methods

Extracted code into helper methods for clarity.
Added addTodo, addDeadline, and addEvent methods to
support adding different task types.

gohjieling834 committed:
Update UITestText for task commands

Updated the input and expected output files to verify all the commands.

Moved Scanner initialisation to main() instead of creating a new Scanner
in getUserInput() to prevent NoSuchElementException during tests.

gohjieling834 committed:
Refactor: improve code quality in all classes

- Added extractDetails method to reduce duplications when adding tasks
- Changed Scanner to a constant
- Renamed argument 'command' in add task methods
  and toggleTaskStatus method to 'userInput' for clarity
- Changed variables in Deadline and Event classes from 'private' to ' protected'

W1ndB10w commented on src/main/java/Berry.java:
```diff
@@ -0,0 +1,147 @@
+import java.util.Scanner;
+
+public class Berry {
+    public static final String DIVIDER = "‧⋆ ✧˚₊‧⋆. ✧˚₊‧⋆‧‧⋆ ✧˚₊‧⋆. ✧˚₊‧⋆‧‧⋆ ✧˚₊‧⋆. ✧˚₊‧⋆‧‧⋆ ✧˚₊‧⋆. ✧˚₊‧⋆‧";
```
This divider pattern is beautifully designed.

W1ndB10w commented on src/main/java/Berry.java:
```diff
@@ -0,0 +1,147 @@
+        System.out.println(DIVIDER + "\n");
+    }
+
+    public static void toggleTaskStatus(String userInput, Task[] tasks) {
```
You consistently used CamelCase in function names and variable names.
They are very clear for user to read and understand their usages. Good job!

W1ndB10w commented on src/main/java/Berry.java:
```diff
@@ -0,0 +1,147 @@
+
+    public static int executeCommand(String userInput, Task[] tasks, int taskCount) {
+        String userCommand = extractCommand(userInput);
+        switch (userCommand) {
```
I like the way that you used "switch-case" when writing your code.
They can convey your logic more clearly.
However, you can also consider adding a "**default**" section to make the structure more complete.
You also used the correct coding format. Good job!

W1ndB10w reviewed (commented):
Generally clear coding format and good usage of "switch-case".

gohjieling834 committed:
Add BerryException class

Add BerryException class to represent exceptions specific to Berry

gohjieling834 committed:
Update code to handle errors

gohjieling834 committed:
Merge branch 'branch-Level-5'

gohjieling834 committed:
Move classes into packages

gohjieling834 committed:
Fix bugs

Bug 1: When entering an invalid task,
taskCount was still incremented. This caused the list to appear
non-empty and the 'list' command to print null tasks.

Bug 2: Extra spaces in deadline/event date/time input were
preserved and displayed in the task list. These are now trimmed.

gohjieling834 committed:
Merge branch 'branch-A-Packages'

roshnidaksha commented on src/main/java/berry/Berry.java:
```diff
@@ -0,0 +1,179 @@
+
+        printHelloMessage();
+        while (true) {
+            userInput = getUserInput(IN);   //get command from user
```
Avoid giving comments to code that is self-explanatory.

roshnidaksha commented on src/main/java/berry/task/Task.java:
```diff
@@ -0,0 +1,44 @@
+package berry.task;
+
+public class Task {
```
Since Task itself is never directly instantiated and objects are always Todo, Deadline, or Event, it would be better to declare Task as abstract.

roshnidaksha commented on src/main/java/berry/Berry.java:
```diff
@@ -0,0 +1,179 @@
+        } catch (NullPointerException e) {
+            printErrorMessage("This task number does not exist! :|");
+        }
+    }
+}
```
You can consider extracting related data/methods into separate classes to make this class shorter.

roshnidaksha reviewed (commented):
There were a few obvious comments mentioned in the code.
Otherwise, well written and neat code!

gohjieling834 committed:
Update code to use ArrayList for tasks

gohjieling834 committed:
Remove variable taskCount

As taskCount can be replaced with tasks.size(),
it was removed.

gohjieling834 committed:
Add deleteTask method

gohjieling834 committed:
Add file operation methods

gohjieling834 committed:
Merge branch 'branch-Level-6'

gohjieling834 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/berry/Berry.java

gohjieling834 committed:
Update deleteTask to call updateFile

gohjieling834 committed:
Create Jar File

gohjieling834 committed:
Improve code readability

Extracted loadData and printHelloMessage into
start method.

gohjieling834 committed:
Extract all command methods

Extracted all command methods (add, list, delete etc)
and put them in berry.command package to improve readbility
and make code more OOP.

gohjieling834 committed:
Refactor command classes

- Changed the method names in the command classes to 'execute'
- Removed redundant abstract AddCommand class

gohjieling834 committed:
Refactor Berry class

Extracted print methods and getUserInput from Berry class
and put them in Ui class, which is in the berry.ui package,
to make code more OOP.

gohjieling834 committed:
Refactor Berry class

Extracted file operation methods from Berry class
into Storage class, which is in berry.storage package,
to make code more OOP.

gohjieling834 committed:
Extract parsing methods

Extract parsing methods from Berry class and
put them into Parser class.

gohjieling834 committed:
Extract parsing methods from Berry

Extracted parsing methods in Berry class and put it in Parser class,
which is in berry.parser package, to make the code more OOP.

gohjieling834 committed:
Change the main logic of Berry

- Changed the main logic of Berry, by adding a constructor and a run method
- Added TaskList class
- Added ExitCommand class

gohjieling834 committed:
Update extractCommand to use switch

gohjieling834 committed:
Fix bug in Parser class

Added a switch case for command 'unmark'

gohjieling834 committed:
Merge branch 'branch-improve-code'

gohjieling834 committed:
Add FindCommand class

gohjieling834 committed:
Improve code quality

gohjieling834 committed:
Merge branch 'branch_Level_9'

gohjieling834 committed:
Add JavaDoc comments

Added JavaDoc comments to most methods and classes.

gohjieling834 committed:
Merge pull request #1 from gohjieling834/branch-A-JavaDoc

Add JavaDoc comments

gohjieling834 committed:
Update execute in FindCommand to handle no task error

Updated the execute method in FindCommand class to handle the error
when no tasks containing the keyword are found.

gohjieling834 committed:
Add if else case to support mark/unmark errors

Added additional if else cases to handle the event of marking a marked
task or unmarking a unmarked task.

gohjieling834 committed:
Update docs/README.md

gohjieling834 committed:
Merge pull request #2 from gohjieling834/branch-A-UserGuide

Update docs/README.md

gohjieling834 committed:
Merge pull request #3 from gohjieling834/branch-handle-errors

Branch handle errors

gohjieling834 committed:
Update docs/README.md

Included a screenshot of the product and added a
note for the command format.

gohjieling834 committed:
Merge branch 'branch-edit-UserGuide'

gohjieling834 committed:
Remove unused imports

Removed unused imports from AddEventCommand.java
and AddTodoCommand.java.

gohjieling834 committed:
Refactor DeleteCommand.java

- Renamed variable `userInput` to `number` for better clarity, as it
  directly represents the task number to be deleted.
- Replaced `userCommand` with a constant to avoid using a magic value
  directly in the code.
- Improves code readability and maintainability by aligning variable
  names with their actual purpose.

gohjieling834 committed:
Refactor MarkCommand.java

- Renamed variable `userInput` to `number` for better clarity, as it
  directly represents the task number to be marked/unmarked.
- Replaced `userCommand` with a constant to avoid using a magic value
  directly in the code.
- Added a variable `isMarkCommand` to differentiate between mark
  and unmark command.
- Extracted the code in the if else cases into 2 methods, `markTask` and
  `unmarkTask`, reducing the if else cases.

gohjieling834 committed:
Update Parser.java

Changed the arguments passed when creating a MarkCommand
and a DeleteCommand, so that it matches the arguments needed.

gohjieling834 committed:
Refactor Ui.java

- Removed the separate printListMessage() method since it was
  only used inside printList().
- Inlined the print call directly to simplify code.

gohjieling834 committed:
Remove dead code in Task.java

Removed the setDescription() method, since it is never called.

gohjieling834 committed:
Add JavaDoc comments for most constructors

gohjieling834 committed:
Merge branch 'branch-improve-code-v2'

gohjieling834 committed:
Change divider used

Changed the divider used, so that it can render properly.

gohjieling834 committed:
Add line breaks

Added line breaks for some of the messages,
so that it fits nicely under the divider when displayed.

gohjieling834 committed:
Update docs/README.md

Updated the README.md in docs to reflect the new divider
and the new product screenshot.

gohjieling834 committed:
Surround switch case with try catch

Surrounded the switch case in Parser.java with try catch
to handle invalid commands. And updated the JavaDoc
comment accordingly.

gohjieling834 committed:
Update AddDeadlineCommand and AddEventCommand

Added if else cases to handle invalid command formats for deadline and
event commands. And updated the JavaDoc comments accordingly.

gohjieling834 committed:
Update JavaDoc comments in AddTodoCommand.java

The exceptions explanation in the JavaDoc comments were wrong,
so i updated it to the correct ones.

gohjieling834 committed:
Update EXPECTED.TXT

Updated the file to reflect the new divider.

gohjieling834 committed:
Update AddDeadlineCommand.java

- Added 2 methods to handle the checking if the details are valid,
  so that it can handle the error when the details are invalid.

gohjieling834 committed:
Update AddEventCommand.java

- Added 2 new methods to check if details are valid and handle
  invalid details.

gohjieling834 committed:
Fix bug in FindCommand.java

Fixed the bug that occurs when entering only spaces as the keyword.
In this commit, the code now treats spaces for a keyword as invalid.
